packages/api/src/platforms/vtex/utils/shouldUpdateShippingData.ts

```diff
@@ -0,0 +1,50 @@
+import { IStoreSession } from '../../../__generated__/schema'
+import { OrderForm } from '../clients/commerce/types/OrderForm'
+
+export const shouldUpdateShippingData = (
+  orderForm: OrderForm,
+  session: IStoreSession
+) => {
+  if (
+    orderForm.items.length === 0 ||
+    !orderForm.shippingData ||
+    !orderForm.shippingData.logisticsInfo ||
+    orderForm.shippingData.logisticsInfo.length === 0
+  ) {
+    return false
+  }
+
+  const { address, logisticsInfo } = orderForm.shippingData
+
+  const hasDifferentPostalCode =
+    typeof session.postalCode === 'string' &&
+    address?.postalCode !== session.postalCode
+  const hasDifferentGeoCoordinates =
+    typeof session.geoCoordinates === 'object' &&
+    typeof session.geoCoordinates?.latitude === 'number' &&
+    typeof session.geoCoordinates?.longitude === 'number' &&
+    (address?.geoCoordinates[0] !== session.geoCoordinates?.longitude ||
+      address?.geoCoordinates[1] !== session.geoCoordinates?.latitude)
```

Comment:
Same thing here. I think this could be simpler if we created a variable for each operand.

Reply:
Done!